### PR TITLE
[Bug][mongo] Fix MongoDB's rewriteOutputBuffer to emit +I rather than +U(or -U)

### PR DESCRIPTION
**Pull request by loserwang1024**
Fix [#2759](https://github.com/ververica/flink-cdc-connectors/issues/2759)

### CODE REVIEW

**Comment by loserwang1024** [comment]
@Jiabao-Sun , @Shawn-Hx , CC

**Comment by Jiabao-Sun** [comment]
+U and +I should have the same processing in the external system. 
Please give an example to explain the specific impact?

**Comment by loserwang1024** [comment]
@Jiabao-Sun , moreover, if `enableFullDocPrePostImage` = true, also will use MongoDBConnectorFullChangelogDeserializationSchema and  emit -U without +I before.

**Comment by Jiabao-Sun** [comment]
> @Jiabao-Sun , moreover, if `enableFullDocPrePostImage` = true, also will use MongoDBConnectorFullChangelogDeserializationSchema and emit -U without +I before.

-U and +U contains the same key, it should be covered by +U ?

**Comment by loserwang1024** [comment]
> -U and +U contains the same key, it should be covered by +U ?


@Jiabao-Sun , `MongoDBFetchTaskContext#rewriteOutputBuffer` method just covered with modify opration(it may contains +U or -U, depending on config `"scan.full-changelog"`).  Then  `MongoDBConnectorDeserializationSchema#deserialize` will only get +U from it , while `MongoDBConnectorFullChangelogDeserializationSchema#deserialize` will get +U and -U from it

**Comment by Shawn-Hx** [comment]
I agree with @loserwang1024. Logically, data before highwatermark are snapshot data, which should be seen as +I. Otherwise the downstream operators may see a +U/-U record without +I before. I'm not sure whether this behavior may cause data incorrectness or not, but using +I is correct with no doubt. @Jiabao-Sun

**Comment by Jiabao-Sun** [comment]
OK, @loserwang1024 could you help add some tests for this?

**Comment by loserwang1024** [comment]
@Jiabao-Sun , thanks a lot. It's not easy to add test now. 
In https://github.com/ververica/flink-cdc-connectors/issues/2759, I add hooks to do database operations during snapshot split code and also add test in it. However, that PR depends on this PR.Therefore, please merge this PR at first.
The detail can see this branch: https://github.com/loserwang1024/flink-cdc-connectors/tree/add-snapshot-hooks.  MongoDBFullChangelogITCase#testEnableBackfillWithPreHighWaterMark、testEnableBackfillWithPostLowWaterMark is what you need.

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBFetchTaskContext.java [hunk 1]
```diff
@@ -162,6 +163,7 @@ public void rewriteOutputBuffer(
                 case INSERT:
                 case UPDATE:
                 case REPLACE:
+                    value.put(MongoDBEnvelope.OPERATION_TYPE_FIELD, INSERT.getValue());
```
Comment (Jiabao-Sun) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBFetchTaskContext.java:166]:
```suggestion
                    value.put(MongoDBEnvelope.OPERATION_TYPE_FIELD, OperationType.INSERT.getValue());
```

Comment (loserwang1024) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/source/reader/fetch/MongoDBFetchTaskContext.java]: Thanks a lot , I will do it

**Review by Jiabao-Sun — APPROVED** [review]

**Pull request merged**